examples/sensor-control/api_how_to.h

```diff
@@ -0,0 +1,453 @@
+        rs2::context ctx;
+
+        // Using the context we can get all connected devices in a device list
+        rs2::device_list devices = ctx.query_devices();
+
+        rs2::device selected_device;
+        if (devices.size() == 0)
+        {
+            std::cerr << "No device connected, please connect a RealSense device" << std::endl;
+                
+            //To help with the boilerplate code of waiting for a device to connect
+            //The SDK provides the rs2::device_hub class
+            rs2::device_hub device_hub(ctx);
+
+            //Using the device_hub we can block the program until a device connects
+            selected_device = device_hub.wait_for_device();
+        }
+        else
+        {
+            std::cout << "Found the following devices:\n" << std::endl;
+
+            // device_list is a "lazy" container of devices which allows
+            //The device list provides 2 ways of iterating it
+            //The first way is using an iterator (in this case hidden in the Range-based for loop)
+            int index = 0;
+            for (rs2::device device : devices)
+            {
+                std::cout << "  " << index++ << " : " << get_device_name(device) << std::endl;
+            }
+
+            uint32_t selected_device_index = get_user_selection("Select a device by index: ");
+
+            // The second way is using the subscript ("[]") operator:
+            if (selected_device_index >= devices.size())
+            {
+                throw std::out_of_range("Selected device index is out of range");
+            }
+
+            // Update the selected device 
+            selected_device = devices[selected_device_index];
+        }
+
+        return selected_device;
+    }
+
+    static void print_device_information(const rs2::device& dev)
+    {
+        // Each device provides some information on itself
+        // The different types of available information are represented using the "RS2_CAMERA_INFO_*" enum
+
+        std::cout << "Device information: " << std::endl;
+        //The following code shows how to enumarate all of the RS2_CAMERA_INFO
+        //Note that all enum types in the SDK start with the value of zero and end at the "*_COUNT" value
+        for (int i = 0; i < static_cast<int>(RS2_CAMERA_INFO_COUNT); i++)
+        {
+            rs2_camera_info info_type = static_cast<rs2_camera_info>(i);
+            //SDK enum types can be streamed to get a string that represents them
+            std::cout << "  " << std::left << std::setw(20) << info_type << " : ";
+
+            //A device might not support all types of RS2_CAMERA_INFO.
+            //To prevent throwing exceptions from the "get_info" method we first check if the device supports this type of info
+            if (dev.supports(info_type))
+                std::cout << dev.get_info(info_type) << std::endl;
+            else
+                std::cout << "N/A" << std::endl;
```

Comment:
Print the attribute that does not provide info

Reply:
it is printed before the `if`